### PR TITLE
[TikTok] Pass cookies to mobile API

### PR DESCRIPTION
**Pull request by MinePlayersPE**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.
Allows someone to use their cookies to download private videos

### CODE REVIEW

File: yt_dlp/extractor/tiktok.py [hunk 1]
```diff
@@ -206,7 +211,11 @@ def extract_addr(addr, add_meta={}):
             'timestamp': int_or_none(aweme_detail.get('create_time')),
             'formats': formats,
             'thumbnails': thumbnails,
-            'duration': int_or_none(traverse_obj(video_info, 'duration', ('download_addr', 'duration')), scale=1000)
+            'duration': int_or_none(traverse_obj(video_info, 'duration', ('download_addr', 'duration')), scale=1000),
+            'availability': self._availability(
+                is_private='Private' in labels,
+                needs_subscription='Friends only' in labels,
+                is_unlisted='Followers only' in labels)
```
Comment (MinePlayersPE) [inline yt_dlp/extractor/tiktok.py:218]: Is this correct?

Comment (MinePlayersPE) [reply yt_dlp/extractor/tiktok.py:218]: Context:
image

Comment (pukkandan) [reply yt_dlp/extractor/tiktok.py:218]: not really. But unless we want to add new cases, ig this is a good approximation

File: yt_dlp/extractor/tiktok.py [hunk 2]
```diff
@@ -68,6 +69,9 @@ def _call_api(self, ep, query, video_id, fatal=True,
             'cp': 'cbfhckdckkde1',
         }
         self._set_cookie(self._API_HOSTNAME, 'odin_tt', ''.join(random.choice('0123456789abcdef') for _ in range(160)))
+        webpage_cookies = self._get_cookies(self._WEBPAGE_HOST)
+        if webpage_cookies and webpage_cookies.get('sid_tt'):
+            self._set_cookie(self._API_HOSTNAME, 'sid_tt', webpage_cookies.get('sid_tt').value)
```
Comment (pukkandan) [inline yt_dlp/extractor/tiktok.py:74]:
```suggestion
        if webpage_cookies.get('sid_tt'):
            self._set_cookie(self._API_HOSTNAME, 'sid_tt', webpage_cookies['sid_tt'].value)
```
_get_cookies alreays returns a dict

**Comment by iw0nderhow** [comment]
This seems to have broken the extractor for me: I always get a `TypeError` at line 216.

```
PS> .\yt-dlp.cmd -v https://www.tiktok.com/@surrealstation/video/7041764961739148590
[debug] Command-line config: ['-v', 'https://www.tiktok.com/@surrealstation/video/7041764961739148590']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2021.12.01 [91f071af6] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 46383212b
[debug] Python version 3.10.0 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] exe versions: ffmpeg 4.3.1-2020-11-19-full_build-www.gyan.dev, ffprobe 4.3.1-2020-11-19-full_build-www.gyan.dev
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@surrealstation/video/7041764961739148590
[TikTok] 7041764961739148590: Downloading video details
[debug] Sort order given by extractor: quality, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, hdr:12(7), asr, proto, vext, aext, hasaud, source, id
ERROR: argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "C:\Users\iw0\git\yt-dlp\yt_dlp\YoutubeDL.py", line 1335, in wrapper
  File "C:\Users\iw0\git\yt-dlp\yt_dlp\YoutubeDL.py", line 1404, in __extract_info
    ie_result = ie.extract(url)
  File "C:\Users\iw0\git\yt-dlp\yt_dlp\extractor\common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "C:\Users\iw0\git\yt-dlp\yt_dlp\extractor\tiktok.py", line 361, in _real_extract
    return self._extract_aweme_app(video_id)
  File "C:\Users\iw0\git\yt-dlp\yt_dlp\extractor\tiktok.py", line 355, in _extract_aweme_app
    return self._parse_aweme_video_app(aweme_detail)
  File "C:\Users\iw0\git\yt-dlp\yt_dlp\extractor\tiktok.py", line 216, in _parse_aweme_video_app
    is_private='Private' in labels,
TypeError: argument of type 'NoneType' is not iterable
```

This does not occur using the previous `tiktok.py` (from b28cdcc):

```
[debug] Command-line config: ['-v', 'https://www.tiktok.com/@surrealstation/video/7041764961739148590', '--force-overwrites']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, err utf-8, pref cp1252
[debug] yt-dlp version 2021.11.10.1 (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: b28cdcc0e
[debug] Python version 3.10.0 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] exe versions: ffmpeg 4.3.1-2020-11-19-full_build-www.gyan.dev, ffprobe 4.3.1-2020-11-19-full_build-www.gyan.dev
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@surrealstation/video/7041764961739148590
[TikTok] 7041764961739148590: Downloading video details
[debug] Sort order given by extractor: quality, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, hdr:12(7), asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 7041764961739148590: Downloading 1 format(s): bytevc1_720p_280160-2
Deleting existing file #duet with @mr.shitposts #PlutoTVDecades [7041764961739148590].mp4
[debug] Invoking downloader on "https://v39-eu.tiktokcdn.com/3661e01e8753cae810fc6f22595fd3b5/61ba45ec/video/tos/useast5/tos-useast5-ve-0068c003-tx/3c4864e10128414db448435c1ac3e2ce/?a=1180&br=546&bt=273&cd=0%7C0%7C0&ch=0&cr=3&cs=2&cv=1&dr=0&ds=3&er=&ft=SE~VrhrV6_cyq4f.7P-Q5NWSjVQ&l=202112151345390102450151361F56D9BB&lr=all&mime_type=video_mp4&net=0&pl=0&qs=14&rc=M2VtcmQ6ZnFvOTMzZzczNEApZWU7ZGQ6ZWU5NzU8ZTloaWduYnIwcjRfc3BgLS1kMS9zc2I2NmIuNC02LzFjNDA0MWA6Yw%3D%3D&vl=&vr="
[download] Destination: #duet with @mr.shitposts #PlutoTVDecades [7041764961739148590].mp4
[download] 100% of 338.20KiB in 00:00
```

**Comment by MinePlayersPE** [comment]
@iw0nderhow so I may or may not have *only* tested this PR with private videos. Sorry 😅

**Pull request merged**